src/Storages/MergeTree/MergeTreeDataSelectExecutor.cpp

```diff
@@ -1656,6 +1657,14 @@ MarkRanges MergeTreeDataSelectExecutor::filterMarksUsingIndex(
     /// this variable is stored to avoid reading the same granule twice.
     MergeTreeIndexGranulePtr granule = nullptr;
     size_t last_index_mark = 0;
+
+    PostingsCacheForStore cache_in_store;
+
+    if (dynamic_cast<const MergeTreeIndexGinFilter *>(&*index_helper) != nullptr)
+    {
+        cache_in_store.store = GinIndexStoreFactory::instance().get(index_helper->getFileName(), part->data_part_storage);
```

Comment:
Is there a chance that the cache_in_store can grow too large and lead to OOM?

Reply:
We treat scalability as one of our top goals during architectural design. According to our tests, the memory usage is pretty stable. The main memory usage is caching term dictionaries(which has been optimized by FST techniques). For example, for a size of 20GB dataset(hacker news), memory usage is kept around 5 GB. We will further optimize memory usage in the near future.